WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRowViewModel.swift

```diff
@@ -63,6 +63,92 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
     ///
     private let variationDisplayMode: VariationDisplayMode?
 
+    /// Subscription settings extracted from product meta data for a Subscription-type Product, if any
+    ///
+    private(set) var productSubscriptionDetails: ProductSubscription?
+
+    /// Description of the subscription billing details for a Subscription-type Product
+    /// eg: "$60.00 / 2 months"
+    ///
+    var subscriptionBillingDetailsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        guard let subscriptionPrice = productSubscriptionDetails?.price,
+              let subscriptionInterval = productSubscriptionDetails?.periodInterval,
+              let subscriptionPeriod = productSubscriptionDetails?.period,
+              let formattedPrice = currencyFormatter.formatAmount(subscriptionPrice, with: currency) else {
+            return ""
+        }
+
+        let subscriptionFrequency = {
+            switch subscriptionInterval {
+            case "1":
+                return subscriptionPeriod.descriptionSingular
+            default:
+                return subscriptionPeriod.descriptionPlural
+            }
+        }
+        return String.localizedStringWithFormat(Localization.formattedProductSubscriptionBilling,
+                                                formattedPrice, subscriptionInterval, subscriptionFrequency())
+    }
+
+    /// Description of the subscription conditions for a Subscription-type Product
+    /// These are separate from each other, a subscription could have any, all, or none
+    /// eg: "$25.00 signup · 1 month free"
+    ///
+    var subscriptionConditionsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        // Signup fees
+        guard let signUpFee = productSubscriptionDetails?.signUpFee else {
+            return ""
+        }
+        var formattedSignUpFee: String = ""
+        if signUpFee.isEmpty {
+            formattedSignUpFee = ""
+        } else {
+            formattedSignUpFee = currencyFormatter.formatAmount(signUpFee, with: currency) ?? ""
+        }
+
+        // Trial periods
+        guard let trialLength = productSubscriptionDetails?.trialLength,
+              let trialPeriod = productSubscriptionDetails?.trialPeriod else {
+            return ""
+        }
+
+        let formattedTrialDetails = {
+            switch trialLength {
+            case "":
+                // The API allows an empty value for trial length, with a non-nil trial period.
+                // eg: every -empty- days
+                return ""
+            case "0":
+                // The API allows to input a 0-length trial length, with a non-nil trial period.
+                // eg: every zero days
+                return "0"
```

Comment:
Nit: can we treat these two cases the same way?

```suggestion
            case "", "0":
                // The API allows an empty value or 0-length for trial length, with a non-nil trial period.
                // eg: every -empty- days
                return ""
```

Reply:
That's something that I also raised, and seems to be a "feature". Some merchants might want to specifically show "0 signup fees" as a reinforcement of their subscription pricing. We might simplify it in the future, but for now we'll treat them differently.

Reply:
I suggested that because it seems that you consider both cases as `hasNoFreeTrial` below by checking `formattedTrialDetails` with an empty string or "0". Separate the cases here seems redundant to me.